jxl/src/headers/frame_header.rs

```diff
@@ -430,22 +451,13 @@ impl FrameHeader {
         const BLOCK_DIM: u32 = 8;
         self.group_dim() * BLOCK_DIM
     }
-
     fn group_counts(&self) -> (usize, usize) {
         const GROUP_DIM: usize = 256;
         const BLOCK_DIM: usize = 8;
-        const H_SHIFT: [usize; 4] = [0, 1, 1, 0];
-        const V_SHIFT: [usize; 4] = [0, 1, 0, 1];
-        let mut maxhs = 0;
-        let mut maxvs = 0;
-        for ch in self.jpeg_upsampling {
-            maxhs = maxhs.max(H_SHIFT[ch as usize]);
-            maxvs = maxvs.max(V_SHIFT[ch as usize]);
-        }
         let xsize = self.width as usize;
         let ysize = self.height as usize;
-        let xsize_blocks = xsize.div_ceil(BLOCK_DIM << maxhs) << maxhs;
-        let ysize_blocks = ysize.div_ceil(BLOCK_DIM << maxvs) << maxvs;
+        let xsize_blocks = xsize.div_ceil(BLOCK_DIM << self.maxhs) << self.maxhs;
```

Comment:
For later: might now make sense to move this logic to `num_groups`/`num_dc_groups` and drop `group_counts`

Reply:
done, see #50